plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -642,21 +648,324 @@ Invoker createInvoker() {
+            actionableMsg.append(" and/or move logic to a Jenkins Shared Library");
+        if (xMsgStart.length() > 0) {
+            actionableMsg
+                    .append(":\n-----\n")
+                    .append(xMsgStart.toString());
+        }
+        if (overflowedClassNameBreadcrumbs.length() > 0) {
+            actionableMsg
+                    .append("\nGroovy code trail (mentions of pipeline WorkflowScript and/or your JSL in larger stack trace):\n")
+                    .append(overflowedClassNameBreadcrumbs);
+        }
+        if (xMsgStart.length() > 0) {
+            if (!(actionableMsg.substring(actionableMsg.length() - 1).equals("\n")))
+                actionableMsg.append("\n");
+            actionableMsg.append("-----");
+        }
+
+        // Make a full note in server log
+        METHOD_TOO_LARGE_LOGGER.log(Level.FINER,
+                "CpsFlowExecution.reportSuspectedMethodTooLarge: full original Throwable message:\n"
+                        + xStr);
+
+        if (ecCount > 1) {
+            // Not squashing with explicit MethodTooLargeException
+            // re-thrown below, in this codepath we have other errors.
+            return new RuntimeException(actionableMsg.toString(), x);
+        }
+
+        // ecCount == 1 exactly, this is the only problem we saw.
+        // Do not confuse pipeline devs by a wall of text in the
+        // build console, but let the full context be found in
+        // server log with some dedication. Note it is seen at
+        // a different logging verbosity level.
+        METHOD_TOO_LARGE_LOGGER.log(Level.FINE,
+                "CpsFlowExecution.reportSuspectedMethodTooLarge: detected details of MethodTooLargeException:\n"
+                        + mtlEx.getMessage());
+
+        actionableMsg
+                .append("\nComplete details can be seen in server log at FINE/FINER level ")
+                .append("(Jenkins admin access for ")
+                .append(METHOD_TOO_LARGE_LOGGER.getName())
+                .append(" is required)");
+
+        //return new RuntimeException(actionableMsg.toString(), mtlEx);
+        RuntimeException rtex = new RuntimeException(actionableMsg.toString(), null);
+
+        // Avoid having a huge stack trace leading to this pretty log-printer
+        // in the build log.
+        // Technically, ContinuationGroup.fixupStackTrace() uses common
+        // parts of the "real" and CPS-caller stack traces to inject the
+        // async call parts, or skips the hassle if the two stack trace
+        // lists have different "roots" (as non-trivially defined in
+        // ContinuationGroup.hasSameRoots() method).
+        StackTraceElement[] shortStack = new StackTraceElement[0];
+        rtex.setStackTrace(shortStack);
+
+        // Considered passing original context,
+        // but it is shown by ultimate job log :(
+        //rtex.addSuppressed(mtlEx);
+
+        return rtex;
+    }
```

Comment:
I am just going to note the irony of a 285-line method here. :grin:

Reply:
Noted :)